cmd/forwarder/auth.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha512"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-redis/redis"
+	metrics "github.com/rcrowley/go-metrics"
+	log "github.com/sirupsen/logrus"
+)
+
+func newAuth(config AuthConfig, registry metrics.Registry) (*BasicAuth, error) {
+	var err error
+
+	if config.RedisUrl != "" && config.RedisKey == "" {
+		err = errors.New("RedisKey must be set if RedisUrl is set")
```

Comment:
To answer your question: Because writing extraneous assignment lines is something that devs do when they are new to a language. 

I'll take your suggestion and return the error without the assignment.